runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPlugin.java

```diff
@@ -49,28 +56,38 @@
 )
 public class OpponentInfoPlugin extends Plugin
 {
+	private static final Duration WAIT = Duration.ofSeconds(5);
+
 	@Inject
 	private Client client;
 
 	@Inject
 	private OverlayManager overlayManager;
 
 	@Inject
-	private OpponentInfoOverlay overlay;
+	private OpponentInfoOverlay opponentInfoOverlay;
 
 	@Getter(AccessLevel.PACKAGE)
 	private HiscoreEndpoint hiscoreEndpoint = HiscoreEndpoint.NORMAL;
 
+	@Getter(AccessLevel.PACKAGE)
+	private Actor lastOpponent;
+
+	private Instant lastTime = Instant.now();
```

Comment:
Is this correct? should be null, lastTime being time when class is created do not makes sense at all. Also maybe this could use gameTicks instead? maybe